### PR TITLE
[pytket-quantinuum] Use new /machine api fields

### PR DESCRIPTION
**Pull request by ss2165**
See individual commits

### CODE REVIEW

File: modules/pytket-quantinuum/pytket/extensions/quantinuum/backends/quantinuum.py [hunk 1]
```diff
@@ -575,12 +604,21 @@ def cost(
...
                 + " Try running `backend.get_compiled_circuit` first"
             )
-        if syntax_checker:
-            backend = QuantinuumBackend(syntax_checker, _api_handler=self._api_handler)
-        else:
-            backend = QuantinuumBackend(
-                _infer_syntax_checker(self._device_name), _api_handler=self._api_handler
+
+        try:
+            syntax_checker = (
```
Comment (cqc-melf) [inline modules/pytket-quantinuum/pytket/extensions/quantinuum/backends/quantinuum.py:609]: Are you sure you want to override here the variable?

File: modules/pytket-quantinuum/pytket/extensions/quantinuum/backends/quantinuum.py [hunk 2]
```diff
@@ -185,6 +196,23 @@ def _available_devices(
         jr = res.json()
         return jr  # type: ignore
 
+    @classmethod
+    def _dict_to_backendinfo(cls, dct: Dict[str, Any]) -> BackendInfo:
+        name: str = dct.pop("name")
+        n_qubits: int = dct.pop("n_qubits")
+        gate_set: List[str] = dct.pop("gateset", [])
```
[the remark this answers is not in the document]
Comment (ss2165) [reply modules/pytket-quantinuum/pytket/extensions/quantinuum/backends/quantinuum.py:203]: yes, that is ensured by the `_get_gateset` function

File: modules/pytket-quantinuum/pytket/extensions/quantinuum/backends/quantinuum.py [hunk 3]
```diff
@@ -76,7 +77,6 @@
     OpType.Rz,
     OpType.PhasedX,
     OpType.ZZMax,
-    OpType.ZZPhase,
```
[the remark this answers is not in the document]
Comment (ss2165) [reply modules/pytket-quantinuum/pytket/extensions/quantinuum/backends/quantinuum.py:79]: it is added based on the machine availability later

File: modules/pytket-quantinuum/pytket/extensions/quantinuum/backends/quantinuum.py (hunk 1 above)
Comment (ss2165) [reply modules/pytket-quantinuum/pytket/extensions/quantinuum/backends/quantinuum.py:609]: yeah, I don't see any issue with it

File: modules/pytket-quantinuum/pytket/extensions/quantinuum/backends/quantinuum.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (cqc-melf) [reply modules/pytket-quantinuum/pytket/extensions/quantinuum/backends/quantinuum.py:79]: Perfect, thank you for explaining that.

**Review by cqc-melf — APPROVED** [review]

**Pull request merged**